README.md

```diff
@@ -435,6 +355,347 @@ One App can be started via docker or if built from source by running `node lib/s
+* 🚫 Child Module  
+
+**Shape**
+```js
+RootModule.appConfig = {
+  csp: String,
+};
+```
+
+The `csp` static `String` should be a valid [Content Security Policy (CSP)](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) for your application which will be passed on to the HTML markup rendered by the Browser.
+
+> We recommend using something like [content-security-policy-builder](https://www.npmjs.com/package/content-security-policy-builder) to create your CSP string.
+
+**📘 More Information**
+* Example: [Frank Lloyd Root's CSP](./prod-sample/sample-modules/frank-lloyd-root/0.0.0/src/csp.js)
+* [content-security-policy-builder](https://www.npmjs.com/package/content-security-policy-builder)
+
+##### `corsOrigins`
+**Module Type**
+* ✅ Root Module 
+* 🚫 Child Module  
+
+**Shape**
+```js
+Module.appConfig = {
+  corsOrigins: [String],
+};
+```
+
+The `corsOrigins` directive accepts an array of `String` URL origin domains.
+This will allow requests from those origins to make POST requests to the server.
+
+In practice, this allows POST requests from given origins to return partially rendered Modules.
+
+**📘 More Information**
+* [Frank Lloyd Root's `appConfig`](https://github.com/americanexpress/one-app/blob/master/prod-sample/sample-modules/frank-lloyd-root/0.0.0/src/config.js)
+* In Practice: [Partial Rendering](#partial-rendering)
+
+##### `configureRequestLog`
+**Module Type**
+* ✅ Root Module 
+* 🚫 Child Module  
+
+**Shape**
+```js
+Module.appConfig = {
+  configureRequestLog: ({
+    req, // Express req
+    log, // One App Log Shape
+  }) => ({
+    // returns reshaped log object
+  }),
+};
+```
+
+The `configureRequestLog` directive a callback that takes Express's `req` and One App's `log` object. This allows for customizing the `log` object based on `req` parameters to add additional metadata to the logger.
+
+**Log Shape**
+```
+{
+  type: 'request',
+  request: {
+    direction,
+    protocol,
+    address: {
+      uri,
+    },
+    metaData: {
+      method,
+      correlationId,
+      host,
+      referrer,
+      userAgent,
+      location,
+      forwarded,
+      forwardedFor,
+      locale,
+    },
+    timings: {
+      // See values in https://www.w3.org/TR/navigation-timing/
+    },
+    statusCode,
+    statusText,
+  },
+};
+```
+
+**📘 More Information**
+* Example: [Frank Lloyd Root's `appConfig`](https://github.com/americanexpress/one-app/blob/master/prod-sample/sample-modules/frank-lloyd-root/0.0.0/src/config.js)
+
+##### `extendSafeRequestRestrictedAttributes`
+**Module Type**
+* ✅ Root Module 
+* 🚫 Child Module  
+
+**Shape**
+```js
+Module.appConfig = {
+  extendSafeRequestRestrictedAttributes: {
+    headers: [String],
+    cookies: [String],
+  },
+};
+```
+
+The `extendSafeRequestRestrictedAttributes` directive accepts a list of cookie names in `cookies` and header identifiers in `headers`. These will allow specified cookies and headers to be whitelisted and attached to the Express `req` object.
```

Comment:
The cookies and headers are already attached to the `req` object, this just allows those headers/cookies to be included in the `req`  object which gets parsed to initial state

Reply:
Fixed in 7bc9f58